src/engine/Operation.cpp

```diff
@@ -211,144 +209,95 @@ void Operation::checkTimeout() const {
-    }
-    // Only the result that was actually computed (or read from cache) knows
-    // the correct information about the children computations.
-    _runtimeInfo.children_ = std::move(runtimeInfo->children_);
-  } else {
-    // The result was computed by this operation (not read from the cache).
-    // Therefore, for each child of this operation the correct runtime is
-    // available.
-    _runtimeInfo.children_.clear();
-    for (auto* child : getChildren()) {
-      AD_CONTRACT_CHECK(child);
-      _runtimeInfo.children_.push_back(
-          child->getRootOperation()->getRuntimeInfo());
-    }
+
+  if (wasCached) {
+    _runtimeInfo->originalOperationTime_ = _runtimeInfo->getOperationTime();
   }
   if (_executionContext) {
-    _executionContext->signalQueryUpdate(_runtimeInfo);
+    _executionContext->signalQueryUpdate(*_rootRuntimeInfo);
   }
 }
 
-// ____________________________________________________________________________________________________________________
-void Operation::updateRuntimeInformationOnSuccess(
-    const ConcurrentLruCache ::ResultAndCacheStatus& resultAndCacheStatus,
-    size_t timeInMilliseconds) {
-  updateRuntimeInformationOnSuccess(
-      *resultAndCacheStatus._resultPointer->resultTable(),
-      resultAndCacheStatus._cacheStatus, timeInMilliseconds,
-      resultAndCacheStatus._resultPointer->runtimeInfo());
-}
-
 // _____________________________________________________________________________
 void Operation::updateRuntimeInformationWhenOptimizedOut(
-    std::vector<RuntimeInformation> children,
     RuntimeInformation::Status status) {
-  _runtimeInfo.status_ = status;
-  _runtimeInfo.children_ = std::move(children);
+  _runtimeInfo->status_ = status;
   // This operation was optimized out, so its operation time is zero.
   // The operation time is computed as
   // `totalTime_ - #sum of childrens' total time#` in `getOperationTime()`.
   // To set it to zero we thus have to set the `totalTime_` to that sum.
-  auto timesOfChildren = _runtimeInfo.children_ |
+  auto timesOfChildren = _runtimeInfo->children_ |
                          std::views::transform(&RuntimeInformation::totalTime_);
-  _runtimeInfo.totalTime_ =
+  _runtimeInfo->totalTime_ =
       std::accumulate(timesOfChildren.begin(), timesOfChildren.end(), 0.0);
 
   if (_executionContext) {
-    _executionContext->signalQueryUpdate(_runtimeInfo);
-  }
-}
-
-// _____________________________________________________________________________
-void Operation::updateRuntimeInformationWhenOptimizedOut(
-    RuntimeInformation::Status status) {
-  auto setStatus = [&status](RuntimeInformation& rti,
-                             const auto& self) -> void {
-    rti.status_ = status;
-    rti.totalTime_ = 0;
-    for (auto& child : rti.children_) {
-      self(child, self);
-    }
-  };
-  setStatus(_runtimeInfo, setStatus);
-
-  if (_executionContext) {
-    _executionContext->signalQueryUpdate(_runtimeInfo);
```

Comment:
We have to discuss all the deleted overload, whether they are really unneeded and have to change that much.